src/client.ts

```diff
@@ -71,27 +71,27 @@ export class APIClient {
     const endpoint = this.withBaseUri(
       `artifactory/api/security/users/${this.clientAdminName}`,
     );
+    let response: Response;
     try {
-      const response = await this.request(endpoint, 'GET');
-
-      if (response.status == 404) {
-        // If we got a 404 it is likely that the client admin name isn't valid
-        throw new IntegrationValidationError(
-          "Recieved 404: Verify the 'Client Administrator Name' provided in the Config is a valid user.",
-        );
-      } else if (response.status !== 200) {
-        throw new IntegrationProviderAuthenticationError({
-          endpoint,
-          status: response.status,
-          statusText: response.statusText,
-        });
-      }
+      response = await this.request(endpoint, 'GET');
```

Comment:
This isn't a big deal but it looks like `response` is only set on this line... so you could just do 
```suggestion
      const response = await this.request(endpoint, 'GET');
```

and delete `let response: Response` above. I think it'd also allow you to remove the optional chaining (`response?.status` -> `response.status`) below.

Reply:
This won't work because then the response scope will be restricted to the try block. The way it is now allows it to escape the try block and throw a different error like the 404 one if necessary.

Reply:
omg what a fool I am 🤦